cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -150,6 +177,27 @@ func (k *proxy) NewClient() (client.Client, error) {
 	return c, nil
 }
 
+func (k *proxy) ClusterAvailable() bool {
```

Comment:
I think returning the error might be useful. Otherwise it's harder to debug

Reply:
Are you suggesting  
a) `func (k *proxy) ClusterAvailable() error` 
OR 
b) `func (k *proxy) ClusterAvailable() (bool, error)`

option A is probably okay but option B might not be that simple. The way we check if the cluster is available ie by checking for errors in the underlying function. It might not be very clear to decide when to return an error and when to return a nil.

Reply:
I was thinking about the first one. My idea was that in any case when the cluster is not available there will be some kind of error. If we return the error the user can get more details about what's going wrong. Otherwise he has to reproduce the check with something else.

Reply:
Hm taking a closer look. Couldn't we use:
* `return false, error` in all the cases you currently have `return false`
* `return true, nil` in the case you currently have `return true`

`func (k *proxy) ClusterAvailable() (bool, error)` Is probably the better choice regarding func naming / return parameter

Reply:
> Hm taking a closer look. Couldn't we use:
> * return false, error in all the cases you currently have return false
> * return true, nil in the case you currently have return true
>
> func (k *proxy) ClusterAvailable() (bool, error) Is probably the better choice regarding func naming / return parameter

If false <-> err and true <-> nil always go together then it becomes redundant. From a consumer point of view, the expression then becomes:
```go
# either  
if _, err := proxy.ClusterAvailable(); err != nil { ... }
# or
if available, _ := proxy.ClusterAvailable(); !available { ... }
```
If we rename the function to `func (k *proxy) CheckClusterAvailable() error` then I think the func name makes it okay to return just an error.
WDYT?

Reply:
Agree, that the bool is redundant. I think the signature you suggested is fine.